src/writer.rs

```diff
@@ -63,11 +63,25 @@ impl<W: Write> BzCompressor<W> {
         }
         Ok(self.w.take().unwrap())
     }
+
+    /// Returns the number of bytes produced by the compressor
+    pub fn total_out(&self) -> u64 {
+        self.stream.total_out()
+    }
+
+    /// Returns the number of bytes consumed by the compressor
```

Comment:
Done.  I went ahead and added these to the other stream types.

This brings up an interesting asymmetry in the API: with the reader API, it's impossible (or perhaps just unintuitive) to force it to flush all the available data through.  I added a doc comment to that effect.  I would generally want it to flush when it gets a EOF from the underlying stream, but I don't see anywhere in the code that would cause that (I could just have missed it).